Signal/src/view controllers/MessagesViewController.m

```diff
@@ -2018,8 +2018,15 @@ - (YapDatabaseConnection *)editingDatabaseConnection {
     return _editingDatabaseConnection;
 }
 
-
 - (void)yapDatabaseModified:(NSNotification *)notification {
+    // Currently, we update thread and message state every time
+    // the database is modified.  That doesn't seem optimal, but
+    // in practice it's efficient enough.
+
+    // We need to `beginLongLivedReadTransaction` before we update our
+    // models in order to jump to the most recent commit.
+    NSArray *notifications = [self.uiDatabaseConnection beginLongLivedReadTransaction];
```

Comment:
This is the first key change - we need to ensure we're updating the model from the latest database state, so we need to call `beginLongLivedReadTransaction` _before_ we update our thread.